### PR TITLE
fix: use /usr/local/bin as bin path

### PR DESCRIPTION
**Pull request by Chumper**
Changes the bin path from `/opt/buildpack/bin` to `/usr/local/bin` as discussed in #300

### CODE REVIEW

File: src/usr/local/buildpack/utils/filesystem.sh [hunk 1]
```diff
@@ -44,13 +44,17 @@ function setup_directories () {
...
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_env_path)"
   # contains the latest version of the tools
-  mkdir -m 770 "$(get_version_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_version_path)"
   # contains the wrapper and symlinks for the tools
-  mkdir -m 770 "$(get_bin_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 755 "$(get_bin_path)"
```
Comment (viceice) [inline src/usr/local/buildpack/utils/filesystem.sh:57]: Does it change permission when directory exist? As `/usr/local/bin` already exist.

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

**Comment by viceice** [comment]
Merging this as fix later, otherwise we don't get a new build / release.

File: src/usr/local/buildpack/utils/filesystem.sh (hunk 1 above)
Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:57]: It does not, the test failed before when I tested for `770`. So the `-p` checks if the directory exist and if it does simply returns.

Comment (viceice) [reply src/usr/local/buildpack/utils/filesystem.sh:57]: Ok, so we need to make sure it's writable by root group, so we need `775` on it.

File: src/usr/local/buildpack/utils/filesystem.sh [hunk 2]
```diff
@@ -44,13 +44,17 @@ function setup_directories () {
...
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_env_path)"
   # contains the latest version of the tools
-  mkdir -m 770 "$(get_version_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_version_path)"
   # contains the wrapper and symlinks for the tools
-  mkdir -m 770 "$(get_bin_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 775 "$(get_bin_path)"
```
Comment (viceice) [inline src/usr/local/buildpack/utils/filesystem.sh:57]: So this only works on tests, as `usr/local/bin` already exist and still has 755?

Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:57]: yes, we need to assume that the bin folder in the image already has `775` or we need to make sure to have a docker (=integration) test for it.

So in the end we may need to set up the folder in the `install-buildpack`

Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:57]: Let me add an integration test for it and add it to the `install-buildpack` method

Comment (viceice) [reply src/usr/local/buildpack/utils/filesystem.sh:57]: Ok, let's do it in `install-buildpack`

Comment (Chumper) [reply src/usr/local/buildpack/utils/filesystem.sh:57]: I added an integration test, but locally it fails. The bin path is still `755`.
If it fails in the actions as well, then somewhere this is changed back.

**Review by viceice — APPROVED** [review]

**Pull request merged**